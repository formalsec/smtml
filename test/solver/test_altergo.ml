open Smtml
open Smtml_tests

let () =
  Printexc.record_backtrace true;
  assert Altergo_mappings.is_available;
  let module Test_solver_params =
    Test_solver_params.Make (Altergo_mappings.Fresh.Make ()) in
  let module Test_solver = Test_solver.Make (Altergo_mappings.Fresh.Make ()) in
  (* let module Test_optimizer =
     Test_optimizer.Make (Altergo_mappings.Fresh.Make ()) in *)
  let module Test_bv = Test_bv.Make (Altergo_mappings.Fresh.Make ()) in
  (* let module Test_fp = Test_fp.Make (Altergo_mappings.Fresh.Make ()) in *)
  let module Test_lia = Test_lia.Make (Altergo_mappings.Fresh.Make ()) in
  (* let module Test_lra = Test_lra.Make (Altergo_mappings.Fresh.Make ()) in *)
  ()
