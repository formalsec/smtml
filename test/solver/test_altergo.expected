(model
  (w i32 4)
  (x i32 1)
  (y i32 2)
  (z i32 3))